Add a test for the `buttonNumber` and `modifierFlags` properties of `WKNavigationAction`

rr-codes opened the pull request:
#### 81b0097021b1003a3942a1b470bf750a846e3b45
<pre>
Add a test for the `buttonNumber` and `modifierFlags` properties of `WKNavigationAction`
<a href="https://bugs.webkit.org/show_bug.cgi?id=280569">https://bugs.webkit.org/show_bug.cgi?id=280569</a>
<a href="https://rdar.apple.com/136894323">rdar://136894323</a>

Reviewed by Wenson Hsieh.

There isn&apos;t an existing test for the `buttonNumber` and `modifierFlags` properties of `WKNavigationAction`,
so this PR adds one.

To facilitate this, the `MouseEventTestHarness` helper class is factored out to a new file so that it can
be re-used in multiple files.

* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/NavigationAction.mm:
(TEST(WKNavigationAction, UserInputState)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(TEST(iOSMouseSupport, DoNotChangeSelectionWithRightClick)):
(TEST(iOSMouseSupport, RightClickDoesNotShowMenuIfPreventDefault)):
(TEST(iOSMouseSupport, TrackButtonMaskFromTouchStart)):
(TEST(iOSMouseSupport, MouseTimestampTimebase)):
(TEST(iOSMouseSupport, EndedTouchesTriggerClick)):
(TEST(iOSMouseSupport, CancelledTouchesDoNotTriggerClick)):
(TEST(iOSMouseSupport, MouseDidMoveOverElement)):
(TEST(iOSMouseSupport, WebsiteMouseEventPolicies)):
(-[WKTestingTouch locationInView:]): Deleted.
(-[WKTestingTouch setLocationInView:]): Deleted.
(-[WKTestingTouch setTapCount:]): Deleted.
(-[WKTestingTouch setPhase:]): Deleted.
(-[WKTestingTouch phase]): Deleted.
(-[WKTestingTouch tapCount]): Deleted.
(-[WKTestingTouch timestamp]): Deleted.
(-[WKTestingTouch _isPointerTouch]): Deleted.
(TestWebKitAPI::MouseEventTestHarness::MouseEventTestHarness): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseMove): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseDown): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseUp): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseCancel): Deleted.
(TestWebKitAPI::MouseEventTestHarness::activeTouches const): Deleted.
(TestWebKitAPI::MouseEventTestHarness::webView const): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseInteraction const): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, DoNotChangeSelectionWithRightClick)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, RightClickOutsideOfTextNodeDoesNotSelect)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, RightClickDoesNotShowMenuIfPreventDefault)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, TrackButtonMaskFromTouchStart)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseTimestampTimebase)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, EndedTouchesTriggerClick)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, CancelledTouchesDoNotTriggerClick)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseDidMoveOverElement)): Deleted.
(TestWebKitAPI::handleUpdatedSelection): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, SelectionUpdatesBeforeContextMenuAppears)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, DisablingTextIteractionPreventsSelectionWhenShowingContextMenu)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, ShowingContextMenuSelectsEditableText)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, ShowingContextMenuSelectsNonEditableText)): Deleted.
(TestWebKitAPI::simulateEditContextMenuAppearance): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, ContextClickAtEndOfSelection)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, WebsiteMouseEventPolicies)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseInitiallyDisconnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseInitiallyConnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseLaterDisconnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseLaterConnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseAlwaysConnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, BasicPointerInteractionRegions)): Deleted.
* Tools/TestWebKitAPI/cocoa/IOSMouseEventTestHarness.h: Added.
(TestWebKitAPI::MouseEventTestHarness::activeTouches const):
(TestWebKitAPI::MouseEventTestHarness::webView const):
(TestWebKitAPI::MouseEventTestHarness::mouseInteraction const):
* Tools/TestWebKitAPI/cocoa/IOSMouseEventTestHarness.mm: Added.
(-[WKTestingTouch locationInView:]):
(-[WKTestingTouch setLocationInView:]):
(-[WKTestingTouch setTapCount:]):
(-[WKTestingTouch setPhase:]):
(-[WKTestingTouch phase]):
(-[WKTestingTouch tapCount]):
(-[WKTestingTouch timestamp]):
(-[WKTestingTouch _isPointerTouch]):
(TestWebKitAPI::MouseEventTestHarness::MouseEventTestHarness):
(TestWebKitAPI::MouseEventTestHarness::mouseMove):
(TestWebKitAPI::MouseEventTestHarness::mouseDown):
(TestWebKitAPI::MouseEventTestHarness::mouseUp):
(TestWebKitAPI::MouseEventTestHarness::mouseCancel):
* Tools/TestWebKitAPI/cocoa/TestDownloadDelegate.mm:

Canonical link: <a href="https://commits.webkit.org/284425@main">https://commits.webkit.org/284425@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/559bc4d9987f563ac26e5a25581041adca9a669e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69325 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48725 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21998 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73407 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20483 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71442 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56526 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20333 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/55138 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13600 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72391 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44457 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59854 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35616 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41123 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17284 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18860 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63065 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17629 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75119 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13307 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16852 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/62803 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13346 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59937 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62709 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10727 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4338 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10590 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44529 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45603 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46798 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45344 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh reviewed (approved):
Since `IOSMouseEventTestHarness` is still iOS-specific, it seems like `Tools/TestWebKitAPI/iOS` is probably a better place to put it (rather than the more general …/cocoa)

rr-codes committed:
Add a test for the `buttonNumber` and `modifierFlags` properties of `WKNavigationAction`
https://bugs.webkit.org/show_bug.cgi?id=280569
rdar://136894323

Reviewed by Wenson Hsieh.

There isn't an existing test for the `buttonNumber` and `modifierFlags` properties of `WKNavigationAction`,
so this PR adds one.

To facilitate this, the `MouseEventTestHarness` helper class is factored out to a new file so that it can
be re-used in multiple files.

* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/NavigationAction.mm:
(TEST(WKNavigationAction, UserInputState)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(TEST(iOSMouseSupport, DoNotChangeSelectionWithRightClick)):
(TEST(iOSMouseSupport, RightClickDoesNotShowMenuIfPreventDefault)):
(TEST(iOSMouseSupport, TrackButtonMaskFromTouchStart)):
(TEST(iOSMouseSupport, MouseTimestampTimebase)):
(TEST(iOSMouseSupport, EndedTouchesTriggerClick)):
(TEST(iOSMouseSupport, CancelledTouchesDoNotTriggerClick)):
(TEST(iOSMouseSupport, MouseDidMoveOverElement)):
(TEST(iOSMouseSupport, WebsiteMouseEventPolicies)):
(-[WKTestingTouch locationInView:]): Deleted.
(-[WKTestingTouch setLocationInView:]): Deleted.
(-[WKTestingTouch setTapCount:]): Deleted.
(-[WKTestingTouch setPhase:]): Deleted.
(-[WKTestingTouch phase]): Deleted.
(-[WKTestingTouch tapCount]): Deleted.
(-[WKTestingTouch timestamp]): Deleted.
(-[WKTestingTouch _isPointerTouch]): Deleted.
(TestWebKitAPI::MouseEventTestHarness::MouseEventTestHarness): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseMove): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseDown): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseUp): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseCancel): Deleted.
(TestWebKitAPI::MouseEventTestHarness::activeTouches const): Deleted.
(TestWebKitAPI::MouseEventTestHarness::webView const): Deleted.
(TestWebKitAPI::MouseEventTestHarness::mouseInteraction const): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, DoNotChangeSelectionWithRightClick)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, RightClickOutsideOfTextNodeDoesNotSelect)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, RightClickDoesNotShowMenuIfPreventDefault)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, TrackButtonMaskFromTouchStart)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseTimestampTimebase)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, EndedTouchesTriggerClick)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, CancelledTouchesDoNotTriggerClick)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseDidMoveOverElement)): Deleted.
(TestWebKitAPI::handleUpdatedSelection): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, SelectionUpdatesBeforeContextMenuAppears)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, DisablingTextIteractionPreventsSelectionWhenShowingContextMenu)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, ShowingContextMenuSelectsEditableText)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, ShowingContextMenuSelectsNonEditableText)): Deleted.
(TestWebKitAPI::simulateEditContextMenuAppearance): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, ContextClickAtEndOfSelection)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, WebsiteMouseEventPolicies)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseInitiallyDisconnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseInitiallyConnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseLaterDisconnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseLaterConnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, MouseAlwaysConnected)): Deleted.
(TestWebKitAPI::TEST(iOSMouseSupport, BasicPointerInteractionRegions)): Deleted.
* Tools/TestWebKitAPI/cocoa/IOSMouseEventTestHarness.h: Added.
(TestWebKitAPI::MouseEventTestHarness::activeTouches const):
(TestWebKitAPI::MouseEventTestHarness::webView const):
(TestWebKitAPI::MouseEventTestHarness::mouseInteraction const):
* Tools/TestWebKitAPI/cocoa/IOSMouseEventTestHarness.mm: Added.
(-[WKTestingTouch locationInView:]):
(-[WKTestingTouch setLocationInView:]):
(-[WKTestingTouch setTapCount:]):
(-[WKTestingTouch setPhase:]):
(-[WKTestingTouch phase]):
(-[WKTestingTouch tapCount]):
(-[WKTestingTouch timestamp]):
(-[WKTestingTouch _isPointerTouch]):
(TestWebKitAPI::MouseEventTestHarness::MouseEventTestHarness):
(TestWebKitAPI::MouseEventTestHarness::mouseMove):
(TestWebKitAPI::MouseEventTestHarness::mouseDown):
(TestWebKitAPI::MouseEventTestHarness::mouseUp):
(TestWebKitAPI::MouseEventTestHarness::mouseCancel):
* Tools/TestWebKitAPI/cocoa/TestDownloadDelegate.mm:

Canonical link: https://commits.webkit.org/284425@main
